doc/source/whatsnew/v1.1.0.rst

```diff
@@ -969,6 +969,7 @@ Plotting
 - Bug in :meth:`DataFrame.boxplot` and :meth:`DataFrame.plot.boxplot` lost color attributes of ``medianprops``, ``whiskerprops``, ``capprops`` and ``medianprops`` (:issue:`30346`)
 - Bug in :meth:`DataFrame.hist` where the order of ``column`` argument was ignored (:issue:`29235`)
 - Bug in :meth:`DataFrame.plot.scatter` that when adding multiple plots with different ``cmap``, colorbars alway use the first ``cmap`` (:issue:`33389`)
+- Bug in :func:`plot.bar`. Ticks position were assigned by value order instead of using the actual value for numeric, or a smart ordering for string. (:issue:`26186` and :issue:`11465`)
```

Comment:
nitpick, i _think_ better to use `DataFrame.plot.bar` and `Series.plot.bar` to replace `plot.bar`

`:func:plot.bar` doesn't seem right

Reply:
@nrebena  and once done, would suggest to ping @jreback or @WillAyd so that this PR could get more attention and merged faster before the conflict issue ^^